File: python/TemperatureConverter.py

print("This program will convert Celcius to Fahrenheit and vice versa.")
t = input("Which would you like to convert? (C of F) ").strip().lower()
match t:
    case "c":
       c = input("What is your temperature in Celsius? ")
       try: 
           c = float(c)
           f = ((c * 9) / 5) + 32
           print(f"Your temperature in Fahrenheit is {f:g}°")
       except ValueError:
           print("This is not a number.")
    case "f": 
        f = input("What is your temperature in Fahrenheit? ")
        try:
            f = float(f)
            c = ((f - 32) * 5) / 9
            print(f"Your temperature in Celsius is {c:g}°")
        except ValueError:
            print("This is not a number.")
    case _:
        print("This is not a valid temperature type.")

File: javascript/TemperatureConverter.js
const prompt = require('prompt-sync')({ sigint: true });

let c, f;

console.log("This program will convert Celsius to Fahrenheit or vice versa");
let t = prompt("Which would you like to convert to? (C or F) ").trim().toLowerCase();

switch (t) {
case "c":
    c = Number(prompt("What is your temperature in Celcius? "));
    if (isNaN(c)) {
	console.log("This is not a number.");
    } else {
	f = ((c * 9) / 5) + 32;
	console.log(`Your temperature in Fahrenheit is ${f}°.`);
    }
    break;
case "f":
    f = Number(prompt("What is your temperature in Fahrenheit? "));
    if (isNaN(f)) {
	console.log("This is not a number.");
    } else {
	c = ((f - 32) * 5) / 9;
	console.log(`Your temperature in Celsius is ${c}°.`);
    }
    break;
default:
    console.log(`That is not a valid temperature type.`);
}


// c = ((f - 32) * 5) / 9
// f = ((c * 9) / 5) + 32
